HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc

```diff
@@ -155,36 +160,71 @@ void HLTScoutingPFProducer::produce(edm::StreamID sid, edm::Event &iEvent, edm::
             break;
           ++index_counter;
         }
-
-        outPFCandidates->emplace_back(MiniFloatConverter::reduceMantissaToNbitsRounding(cand.pt(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.eta(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.phi(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.mass(), mantissaPrecision),
-                                      cand.pdgId(),
-                                      vertex_index);
+        float normchi2{}, dz{}, dxy{}, dzError{}, dxyError{}, trk_pt{}, trk_eta{}, trk_phi{};
+        uint8_t lostInnerHits{}, quality{};
+        if (doTrackRelVars_) {
+          const auto *trk = cand.bestTrack();
+          if (vertexCollection->empty() or trk == nullptr) {
+            dz = 0;
+            dxy = 0;
+          } else {
+            const reco::Vertex &pv = (*vertexCollection)[0];
+            dz = trk->dz(pv.position());
+            dxy = trk->dxy(pv.position());
+          }
+          normchi2 = trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(std::floor(trk->normalizedChi2()),
+                                                                             mantissaPrecision_)
+                         : MiniFloatConverter::reduceMantissaToNbitsRounding(999, mantissaPrecision_);
+          dzError =
+              trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(dz / trk->dzError(), mantissaPrecision_) : 0;
+          dxyError =
+              trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(dxy / trk->dxyError(), mantissaPrecision_) : 0;
+          lostInnerHits = trk ? btagbtvdeep::lost_inner_hits_from_pfcand(cand) : 0;
+          quality = trk ? btagbtvdeep::quality_from_pfcand(cand) : 0;
+          trk_pt = trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(trk->pt(), mantissaPrecision_) : 0;
+          trk_eta = trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(trk->eta(), mantissaPrecision_) : 0;
+          trk_phi = trk ? MiniFloatConverter::reduceMantissaToNbitsRounding(trk->phi(), mantissaPrecision_) : 0;
+        }
```

Comment:
```suggestion
        float normchi2{0}, dz{0}, dxy{0}, dzError{0}, dxyError{0}, trk_pt{0}, trk_eta{0}, trk_phi{0};
        uint8_t lostInnerHits{0}, quality{0};
        if (doTrackRelVars_) {
          const auto *trk = cand.bestTrack();
          if (trk != nullptr) {

            normchi2 = MiniFloatConverter::reduceMantissaToNbitsRounding(std::floor(trk->normalizedChi2()), mantissaPrecision_);
            lostInnerHits = btagbtvdeep::lost_inner_hits_from_pfcand(cand);
            quality = btagbtvdeep::quality_from_pfcand(cand);
            trk_pt = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->pt(), mantissaPrecision_);
            trk_eta = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->eta(), mantissaPrecision_);
            trk_phi = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->phi(), mantissaPrecision_);

            if (not vertexCollection->empty()) {
              const reco::Vertex &pv = (*vertexCollection)[0];

              dz = trk->dz(pv.position());
              dxy = trk->dxy(pv.position());
              dzError = MiniFloatConverter::reduceMantissaToNbitsRounding(dz / trk->dzError(), mantissaPrecision_);
              dxyError = MiniFloatConverter::reduceMantissaToNbitsRounding(dxy / trk->dxyError(), mantissaPrecision_);
            }
          } else {

            normchi2 = MiniFloatConverter::reduceMantissaToNbitsRounding(999, mantissaPrecision_);
          }
        }
```
This should be equivalent, and would look clearer to me (maybe wait to see if others have comments on this part). By the way, I see that `dz` and `dxy` are kept with full precision; I assumed that's intentional.

Reply:
Thanks, `dz` and `dxy` are not supposed to be full precision.

Reply:
@dsperka regarding this. You asked me to fill the track related variables with just empty values (size 0) if the flag `doTrackRelVars_` is False. However if I leave the variables uninitialized (i.e. `float normchi2 ... etc`) I get the following when compiling:

```
warning: 'normchi2' may be used uninitialized in this function
```

I therefore changed `float normchi2` to `float normchi2{}` (or as Marino `float normchi2{0}`) which initializes it as 0. This however is not the same as you asked for. I am not sure if there is a way to leave the variable uninitialized while avoiding the compilation warning or if initializing it as 0 is OK. Thoughts?

Reply:
I just compared file sizes when the flag is set to True and False. With 100 events the Run3ScoutingParticle branch has file size 335485 (True) and 186017 (False). It is clear then that the compression is still very efficient when initialize to 0. David agreed to keep it initialized (thank you, David, for the offline feedback).

Does anybody have any comments on Marino's suggestion? If not then I can commit it (and add reduced precision for dz and dxy).

Reply:
Thanks again for all the feedback. If I don't hear anything further I'll commit the code at 11:00.